internal/controllers/objectsets/revision_reconciler.go

```diff
@@ -0,0 +1,76 @@
+package objectsets
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const revisionReconcilerRequeueDelay = 10 * time.Second
+
+// revisionReconciler determines the .status.revision number by checking previous revisions.
+type revisionReconciler struct {
+	scheme       *runtime.Scheme
+	newObjectSet genericObjectSetFactory
+	client       client.Client
+}
+
+func (r *revisionReconciler) Reconcile(
+	ctx context.Context, objectSet genericObjectSet,
+) (res ctrl.Result, err error) {
+	if objectSet.GetStatusRevision() != 0 {
+		// .status.revision is already set.
+		return
+	}
+
+	if len(objectSet.GetPrevious()) == 0 {
+		// no previous revision(s) specified, default to revision 1
+		objectSet.SetStatusRevision(1)
+		return
+	}
+
+	// Determine new revision number by inspecting previous revisions:
+	revisions := make([]genericObjectSet, len(objectSet.GetPrevious()))
```

Comment:
Isn't it a little more efficient and less complex to do:
```
latestPreviousRevision := 0
....
if prevObjectSet.GetStatusRevision() > latestPreviousRevision:
    latestPreviousRevision = prevObjectSet.GetStatusRevision()
```

Reply:
Yes, you are right. Change incoming :)